libs/wlgen/wlgen/workload.py

```diff
@@ -217,7 +217,8 @@ def run(self,
 
         # Prepend eventually required cgroup command
         if self.cgroup:
-            self.cgroup_cmd = 'cgroups_run_into {}'.format(self.cgroup)
+            self.cgroup_cmd = '{} cgroups_run_into {}'\
+                    .format(self.target.shutils, self.cgroup)
```

Comment:
This fails because `cgroups_run_into` expects a path to a Cgroup, whereas here it gets the object reference (i.e. `<devlib.module.cgroups.CGroup object at 0x7fcf9c747ad0>`).

You should change `self.cgroup` to `self.cgroup.name`.